### PR TITLE
feat: Support ManagedMediaSource

### PR DESCRIPTION
**Pull request by monyone**
Support ManagedMediaSource if enableManagedMediaSource specified (for Safari)

### CODE REVIEW

File: d.ts/mpegts.d.ts [hunk 1]
```diff
@@ -47,6 +47,11 @@ declare namespace Mpegts {
          * @defaultvalue false
          */
         enableWorker?: boolean;
+        /**
+         * @desc Enable Managed Media Source (unstable for now)
+         * @defaultvalue false
+         */
+        enableManagedMediaSource?: boolean;
```
Comment (xqq) [inline d.ts/mpegts.d.ts:54]: I don't think we need a config field to enable this. We should use it as long as it's detected.

Comment (monyone) [reply d.ts/mpegts.d.ts:54]: OK, i will fix ut,
if only ManagedMediaSource avaliable (iPhone), use ManagedMediaSource.

Comment (monyone) [reply d.ts/mpegts.d.ts]: Fixed!

**Comment by xqq** [comment]
Will merge when ManagedMediaSource becomes stable on iOS

**Comment by xqq** [comment]
I have confirmed that this pull request works on the newest macOS Safari 17.0 (17616.1.27.111.22, 17616).
It should be also working on the upcoming iOS 17.1 Safari.

Will merge with some adjustments. Thank you!

**Pull request merged**